src/Domain/DomainHelpers.cpp

```diff
@@ -503,18 +503,51 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
                       const double x_coord_of_shell_center,
                       const bool use_half_wedges, const double aspect_ratio,
                       const bool use_logarithmic_map,
-                      const ShellWedges which_wedges) noexcept {
+                      const ShellWedges which_wedges,
+                      const size_t number_of_layers) noexcept {
+  ASSERT(not use_half_wedges or which_wedges == ShellWedges::All,
+         "If we are using half wedges we must also be using ShellWedges::All.");
+  ASSERT(number_of_layers == 1 or inner_sphericity == outer_sphericity,
+         "If we are using more than one layer the inner and outer sphericities "
+         "must match.");
+  ASSERT(x_coord_of_shell_center == 0.0 or
+             (not use_half_wedges and aspect_ratio == 1.0),
+         "Only one of: using a non-zero translation, using half wedges, or "
+         "using a non-unity aspect_ratio, can be done at a time.");
+  ASSERT(not use_half_wedges or aspect_ratio == 1.0,
```

Comment:
Just `and` together these last two, since they have the same message and are really parts of the same check.

(I would likely have written `(x_coord_of_shell_center != 0.0) + half_wedges + (aspect_ratio != 1.0) <= 1`, but that may be being too clever.)